webapp/src/cloud.ts

```diff
@@ -696,19 +696,36 @@ export async function aiErrorExplainRequest(
     target: string,
     outputFormat: "tour_json" | "text"
 ): Promise<string | undefined> {
+    const startUrl = `/api/copilot/startexplainerror`;
+    const statusUrl = `/api/copilot/explainerrorstatus`;
 
-    const url = `/api/copilot/explainerror`;
     const data = { lang, code, errors, target, outputFormat };
-    let result: string = "";
-
-    const request = await auth.apiAsync(url, data, "POST");
-    if (!request.success) {
-        throw new StatusCodeError(
-            request.statusCode,
-            request.err || `Unable to reach AI. Error: ${request.statusCode}.\n${request.err}`
-        );
+
+    // Start the request.
+    const queryStart = await auth.apiAsync(startUrl, data, "POST");
+    const resultId = queryStart.resp.resultId;
```

Comment:
Should there still be a check that we have the correct data here? If the start function errors, would that mean that the call to `apiAsync` results in a rejected promise? If so, would we be able to get `resultId`?

Reply:
Good catch, we should still handle errors here like we did before. It got lost in the refactor.